builtin/providers/aws/resource_aws_config_config_rule.go

```diff
@@ -0,0 +1,314 @@
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"source_identifier": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateMaxLength(256),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsConfigConfigRulePut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).configconn
+
+	name := d.Get("name").(string)
+
+	sources := d.Get("source").([]interface{})
+	source := sources[0].(map[string]interface{})
+	sourceInput := configservice.Source{
+		Owner:            aws.String(source["owner"].(string)),
+		SourceIdentifier: aws.String(source["source_identifier"].(string)),
+	}
+	if details, ok := source["source_detail"]; ok {
+		sourceInput.SourceDetails = expandConfigRuleSourceDetails(details.(*schema.Set))
+	}
+
+	ruleInput := configservice.ConfigRule{
+		ConfigRuleName: aws.String(name),
+		Source:         &sourceInput,
+	}
+
+	scopes := d.Get("scope").([]interface{})
+	if len(scopes) > 0 {
+		ruleInput.Scope = expandConfigRuleScope(scopes[0].(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		ruleInput.Description = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("input_parameters"); ok {
+		ruleInput.InputParameters = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("maximum_execution_frequency"); ok {
+		ruleInput.MaximumExecutionFrequency = aws.String(v.(string))
+	}
+
+	input := configservice.PutConfigRuleInput{
+		ConfigRule: &ruleInput,
+	}
+	log.Printf("[DEBUG] Creating AWSConfig config rule: %s", input)
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		_, err := conn.PutConfigRule(&input)
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				if awsErr.Code() == "InsufficientPermissionsException" {
+					// IAM is eventually consistent
+					return resource.RetryableError(err)
+				}
+			}
+
+			return resource.NonRetryableError(fmt.Errorf("Failed to create AWSConfig rule: %s", err))
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	d.SetId(name)
+
+	log.Printf("[DEBUG] AWSConfig config rule %q created", name)
+
+	return resourceAwsConfigConfigRuleRead(d, meta)
+}
+
+func resourceAwsConfigConfigRuleRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).configconn
+
+	out, err := conn.DescribeConfigRules(&configservice.DescribeConfigRulesInput{
+		ConfigRuleNames: []*string{aws.String(d.Id())},
+	})
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "NoSuchConfigRuleException" {
+			log.Printf("[WARN] Config Rule %q is gone (NoSuchConfigRuleException)", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	if len(out.ConfigRules) < 1 {
+		log.Printf("[WARN] Config Rule %q is gone (no rules found)", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	log.Printf("[DEBUG] AWS Config config rule received: %s", out)
+
+	rule := out.ConfigRules[0]
```

Comment:
Can we be sure that `out.ConfigRules[0]` is the correct rule? Is there any way we can get > 1 rule?